mill-scalafix/src/com/goyeau/mill/scalafix/ScalafixCache.scala

```diff
@@ -0,0 +1,24 @@
+package com.goyeau.mill.scalafix
+
+import com.goyeau.mill.scalafix.CoursierUtils
+import coursier.core.Repository
+import scalafix.interfaces.Scalafix
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.jdk.CollectionConverters._
+import scala.ref.SoftReference
+
+private[scalafix] object ScalafixCache {
+
+  private val cache = new ConcurrentHashMap[(String, Seq[Repository]), SoftReference[Scalafix]]
+
+  def getOrElseCreate(scalaVersion: String, repositories: Seq[Repository]) =
```

Comment:
Since `repositories` is provided, should we have it in the cache key as well?

Reply:
We do, don't we?

Reply:
oops 🙈 my bad!